d2bs/kolbot/libs/common/Precast.js

```diff
@@ -152,18 +153,21 @@ var Precast = new function () {
 	};
 
 	this.precastSkill = function (skillId) {
-		var swapped,
-			slot = this.getBetterSlot(skillId);
+		var swapped;
 
-		if (slot !== me.weaponswitch) {
+		if (this.bestSlot[skillId] === undefined) {
```

Comment:
change
`if (this.bestSlot[skillId] === undefined) {`
to
`if (!this.bestSlot[skillId]) {`